### PR TITLE
Modified component positioning to solve rendering inconsistencies between different devices

### PR DESCRIPTION
**Pull request by samryderE**
The positioning of components was inconsistent across different device screen sizes. Modifications to flexbox attributes in the most ancestral container as well as child components was necessary to ensure that renders would be positioned as expected regardless of screen size.

Additionally, version updates to expo as well as other dependencies had to be made in order to run builds in Expo Go, the only way to test shake functionality on an actual device. Be sure to run `npm i`.

If you encounter iOS simulator errors and are unable to boot up a simulator, old simulator caches are likely the culprit. Try system preferences -> storage -> developer -> from here delete any existing xCode simulator caches.

### CODE REVIEW

File: AnimateSvg.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ export default function AnimateSvg({ isShakeTriggered, setIsShakeReady, hasIniti
   });
   const sizeAnim = useRef(new Animated.Value(1)).current;
 
+  const leftPx = width * leftPositionPercentDec;
```
Comment (samryderE) [inline AnimateSvg.js:17]: Dynamic calculation of component's `left` positioning.

File: AnimateQuotes.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-export default function AnimateQuotes({ isShakeTriggered, setIsShakeReady, hasInitialTransitionRun, setHasInitialTransitionRun }) {
+export default function AnimateQuotes({ isShakeTriggered, setIsShakeReady, hasInitialTransitionRun, setHasInitialTransitionRun, height, width }) {
   const [isQuoteAnimationRunning, setIsQuoteAnimationRunning] = useState(false);
   const [quoteIndex, setQuoteIndex] = useState(0);
 
   const fadeAnim = useRef(new Animated.Value(0)).current;
   const sizeAnim = useRef(new Animated.Value(0.5)).current;
 
+  const topPx = height * topPositionPercentDec;
+  const leftPx = width * leftPositionPercentDec;
```
Comment (samryderE) [inline AnimateQuotes.js:15]: Dynamic calculation of component's `left` and `top` attribute values for absolute positioning.

File: constants.js [hunk 3]
```diff
@@ -30,3 +30,7 @@ export const quotes = [
   "protect the people you love. call out bullshit at all costs. appreciate free thought in all its glory.",
   "we only want to believe the 'good' things about the people we admire, when what makes them heroes is how they negotiate their flaws.",
 ];
+
+// Percentages to be multiplied against useWindowDimensions() height and width values for setting left and top attributes of absolute positioned Animate components
+export const leftPositionPercentDec = .06;
+export const topPositionPercentDec = .45;
```
Comment (russellwmoore) [inline constants.js:36]: Change these to the `CAPITAL_SNAKE_CASE` since they are constants. 
Suggestion: const `LEFT_POSITION_PERCENT`

**Comment by russellwmoore** [comment]
See if you can boot up 3 different simulators with very different sizes and capture screen shots of this change working on each

File: AnimateQuotes.js (hunk 2 above)
Comment (russellwmoore) [inline AnimateQuotes.js:15]:
```suggestion
  const topPosition = height * topPositionPercentDec;
  const leftPosition = width * leftPositionPercentDec;
```

File: AnimateSvg.js (hunk 1 above)
Comment (russellwmoore) [reply AnimateSvg.js:17]:
```suggestion
  const leftPosition = width * leftPositionPercentDec;
```

Also, why don't we need a `topPosition` for this component like the other one?

File: src/AnimateQuotes.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 
-export default function AnimateQuotes({ isShakeTriggered, setIsShakeReady, hasInitialTransitionRun, setHasInitialTransitionRun }) {
+export default function AnimateQuotes({ isShakeTriggered, setIsShakeReady, hasInitialTransitionRun, setHasInitialTransitionRun, height }) {
   const [isQuoteAnimationRunning, setIsQuoteAnimationRunning] = useState(false);
   const [quoteIndex, setQuoteIndex] = useState(0);
 
   const fadeAnim = useRef(new Animated.Value(0)).current;
   const sizeAnim = useRef(new Animated.Value(0.5)).current;
 
+  const topPosition = height * TOP_POSITION_PERCENT;
```
Comment (samryderE) [inline src/AnimateQuotes.js:14]: Dictates where component's position will land absolutely in the parent component container upon completion of animation sequences.

File: src/App.js [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { StyleSheet, Text, SafeAreaView, View, useWindowDimensions } from 'react-native';
 import AnimateSvg from './AnimateSvg';
 import AnimateQuotes from './AnimateQuotes';
 import { useIsShake } from './useIsShake';
 
 export default function App() {
   const [hasInitialTransitionRun, setHasInitialTransitionRun] = useState(false);
   const { isShakeTriggered, setIsShakeReady } = useIsShake();
+  const { height } = useWindowDimensions();
```
Comment (samryderE) [inline src/App.js:10]: In mobile devices that can modify app window size in relationship to screen size, `height` will update dynamically.

File: src/App.js [hunk 6]
```diff
@@ -20,6 +21,7 @@ const styles = StyleSheet.create({
   container: {
     flex: 1,
     justifyContent: 'center',
+    alignItems: 'center',
```
Comment (samryderE) [inline src/App.js:24]: Solved inconsistent horizontal positioning issue for child components.

**Review by russellwmoore — APPROVED** [review]
Make a note in the description about the upgrades, then you are good to go

File: package.json [hunk 7]
```diff
@@ -9,13 +9,13 @@
...
+    "expo-image": "~1.12.8",
+    "expo-sensors": "~13.0.5",
+    "expo-status-bar": "~1.12.1",
     "react": "18.2.0",
-    "react-native": "0.73.4",
-    "expo-sensors": "~12.9.1",
-    "expo-image": "~1.10.6",
-    "react-native-svg": "14.1.0"
+    "react-native": "0.74.1",
+    "react-native-svg": "15.2.0-rc.0"
```
Comment (samryderE) [inline package.json:18]: Updated versions. Be sure to run `npm i`.

File: src/AnimateQuotes.js [hunk 8]
```diff
@@ -86,8 +88,10 @@ export default function AnimateQuotes({ isShakeTriggered, setIsShakeReady, hasIn
       opacity: fadeAnim, // Bind opacity to animated value
       transform: [{ scale: sizeAnim }], // Bind transform to animated value
       position: 'absolute',
-      left: '6%',
-      top: '45%',
+      // following line will hold Quotes component to center of window, eliminating the floating upward and downward motion upon fade in and out, respectively
+      // justifyContent: 'center',
+      // following line to be used instead of justifyContent attribute, allowing floating upward and downard effect up fade in and out, respectively
+      top: topPosition,
```
Comment (samryderE) [inline src/AnimateQuotes.js:94]: Options here for how quotes are rendered during animations.

**Pull request merged**